controllers/raycluster_controller.go

```diff
@@ -0,0 +1,320 @@
+		)
+		if err != nil {
+			logger.Error(err, "Error deleting OAuth Route, retrying", logRequeueing, strTrue)
+			return ctrl.Result{RequeueAfter: requeueTime}, nil
+		}
+		return ctrl.Result{}, nil
+	}
+
+	_, err := r.routeClient.Routes(cluster.Namespace).Apply(ctx, desiredClusterRoute(&cluster), metav1.ApplyOptions{FieldManager: controllerName, Force: true})
+	if err != nil {
+		logger.Error(err, "Failed to update OAuth Route")
+	}
+
+	_, err = r.kubeClient.CoreV1().Secrets(cluster.Namespace).Apply(ctx, desiredOAuthSecret(&cluster, r), metav1.ApplyOptions{FieldManager: controllerName, Force: true})
+	if err != nil {
+		logger.Error(err, "Failed to create OAuth Secret")
+	}
+
+	_, err = r.kubeClient.CoreV1().Services(cluster.Namespace).Apply(ctx, desiredOAuthService(&cluster), metav1.ApplyOptions{FieldManager: controllerName, Force: true})
+	if err != nil {
+		logger.Error(err, "Failed to update OAuth Service")
+	}
+
+	_, err = r.kubeClient.CoreV1().ServiceAccounts(cluster.Namespace).Apply(ctx, desiredServiceAccount(&cluster), metav1.ApplyOptions{FieldManager: controllerName, Force: true})
+	if err != nil {
+		logger.Error(err, "Failed to update OAuth ServiceAccount")
+	}
+
+	_, err = r.kubeClient.RbacV1().ClusterRoleBindings().Apply(ctx, desiredOAuthClusterRoleBinding(&cluster), metav1.ApplyOptions{FieldManager: controllerName, Force: true})
+	if err != nil {
+		logger.Error(err, "Failed to update OAuth ClusterRoleBinding")
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func crbNameFromCluster(cluster *rayv1.RayCluster) string {
+	return cluster.Name + "-" + cluster.Namespace + "-auth" // NOTE: potential naming conflicts ie {name: foo, ns: bar-baz} and {name: foo-bar, ns: baz}
+}
+
+func (r *RayClusterReconciler) deleteIfNotExist(ctx context.Context, namespacedName types.NamespacedName, obj client.Object) error {
+	err := r.Client.Get(ctx, namespacedName, obj)
+	if err != nil && errors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+	return r.Client.Delete(ctx, obj, &deleteOptions)
+}
+
+func desiredOAuthClusterRoleBinding(cluster *rayv1.RayCluster) *rbacapply.ClusterRoleBindingApplyConfiguration {
+	return rbacapply.ClusterRoleBinding(
+		crbNameFromCluster(cluster)).
+		WithLabels(map[string]string{"ray.io/cluster-name": cluster.Name}).
+		WithSubjects(
+			rbacapply.Subject().
+				WithKind("ServiceAccount").
+				WithName(cluster.Name).
+				WithNamespace(cluster.Namespace),
+		).
+		WithRoleRef(
+			rbacapply.RoleRef().
+				WithAPIGroup("rbac.authorization.k8s.io").
+				WithKind("ClusterRole").
+				WithName("system:auth-delegator"),
+		).
+		WithOwnerReferences(
+			v1.OwnerReference().WithUID(cluster.UID).WithName(cluster.Name).WithKind(cluster.Kind).WithAPIVersion(cluster.APIVersion),
+		)
+}
+
+func desiredServiceAccount(cluster *rayv1.RayCluster) *coreapply.ServiceAccountApplyConfiguration {
+	return coreapply.ServiceAccount(cluster.Name, cluster.Namespace).
```

Comment:
Just to make sure, is it expected for the service account name and other resources to differ in the values from the SDK? I.e., the name here would be `cluster.Name`, while in the SDK we would have [`cluster.Name-oauth-proxy`](https://github.com/project-codeflare/codeflare-sdk/blob/main/src/codeflare_sdk/utils/generate_yaml.py#L557).

Reply:
this shouldn't really matter since it is just internals and won't create breaking changes for the user but I will adjust them to be consistent